docs/book/src/developer/providers/contracts.md

```diff
@@ -0,0 +1,31 @@
+# Provider contract
+
+Cluster API defines a contract which requires providers to implement certain fields and patterns in their CRDs and controllers. This contract is required for providers to work correctly with Cluster API.
+
+Cluster API defines the following contracts:
+
+- [Infrastructure provider contract](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/cluster.html?highlight=contracts#infrastructure-provider)
+- [Control Plane provider contract](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/control-plane.html#crd-contracts)
+- [Machine provider contract](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/machine.html#cluster-api)
+- [clusterctl provider contract](https://cluster-api.sigs.k8s.io/clusterctl/provider-contract.html?highlight=contracts#clusterctl-provider-contract)
+- [Multi tenancy contract](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/multi-tenancy.html?highlight=contracts#contract)
+
+## API version labels
```

Comment:
I'd like to leave it here right now. I'm planning to create issues around re-doing the "Implementers guide" section, so I'd like to leave it as is until it can be improved holistically. This just matches the way things were before.